internal/rpc/clients.go

```diff
@@ -71,6 +72,40 @@ func GRPCClientFromConfig(cfg config.View, prefix string) (*grpc.ClientConn, err
 	return GRPCClientFromParams(clientParams)
 }
 
+// GRPCClientFromEndpoint creates a gRPC client connection from endpoint.
+func GRPCClientFromEndpoint(cfg config.View, hostname string, port int) (*grpc.ClientConn, error) {
```

Comment:
I'm going to approve but I think this code is doing too much.
Add a TODO to investigate keep a cache of the `*tls.Config`.